.github/workflows/cpp.yml

```diff
@@ -20,9 +20,6 @@ name: C++
 
 on:
   push:
-    branches:
-      - master
-      - maint-*
```

Comment:
Note that this will trigger as many builds as we run on apache/arrow's master, which is more than we trigger for pull requests.